src/api/services/translations_cache.py

```diff
@@ -1,51 +1,140 @@
+        translation, err = self.store.get_translation(context, args)
+        return translation if translation else None, err
+
+    def get_target_languages (self, context: Context, target_language_code) -> Tuple[ list, None ]:
+        supported_languages, err = SupportedLanguageStore().list_supported_languages(context, { })
+        return [ (to_third_party_lang_code(language.code), language.code) for language in supported_languages if
+                 language.code != DEFAULT_SOURCE_LANGUAGE_CODE ] if supported_languages else None, err
+
+    def translate_text_field (self, context: Context, field_value: str, target_language_code) -> Tuple[ dict or None, None ]:
+        text_hash, err = self.textHashService.create_text_hash(context, { "text": field_value })
+        target_language, err = self.get_target_languages(context, target_language_code)
+        source_language_code = to_third_party_lang_code(DEFAULT_SOURCE_LANGUAGE_CODE)
+
+        if err:
+            return None, err
+
+        hash = getattr(text_hash, 'hash', None)
+
+        if hash:
+            for target_language_tuple in target_language:
+                translated_text = self.translator.translate_text(text = field_value,
+                                                            source_language =  target_language_tuple[0],
+                                                            target_language = source_language_code
+                                                                 )
+                translation, err = self.create_translation(context, {
+                    "hash": text_hash,
+                    "source_language": DEFAULT_SOURCE_LANGUAGE_CODE,
+                    "target_language": target_language_tuple[1],
+                    "translated_text": translated_text
+                })
+
+                if err:
+                    return None, err
+
+                return translation, None
+
+    def translate_json_field (self, context: Context, value: dict, target_language_code, ignoredKeys: Set) -> Tuple[ dict or None, None ]:
+        for key, field_value in value.items():
+            if isinstance(field_value, str):
+                return self.translate_text_field(context, field_value, target_language_code)
+            elif isinstance(field_value, dict):
+                return self.translate_json_field(context, field_value, target_language_code, ignoredKeys)
+            elif isinstance(field_value, list):
+                for item in field_value:
+                    if isinstance(item, dict):
+                        return self.translate_json_field(context, item, target_language_code, ignoredKeys)
+                    elif isinstance(item, str):
+                        return self.translate_text_field(context, item, target_language_code)
+                    else:
+                        continue
+            else:
+                continue
```

Comment:
there are a bunch of functions here where we return None sometimes and something else another time and its not clear why.  I would like us to avoid that as much as possible.  But if there is a good reason for that, i'd like to hear it.